### PR TITLE
add query option of disabling upsert during query

### PR DESCRIPTION
**Pull request by yupeng9**
## Description
Part of a series of PRs for #4261
Check this [doc](https://docs.google.com/document/d/1qljEMndPMxbbKtjlVn9mn2toz7Qrk0TGQsHLfI--7h8/edit#heading=h.lsfmyoyyxtgt) out for the new design

Add a query option for disabling upsert (mostly for debugging purpose)

```
select event_id, count(*) from meetupRsvp option(disableUpsert=false) group by event_id limit 10
```

displays:
Screen Shot 2020-10-06 at 4 49 06 PM

### CODE REVIEW

**Review by mayankshriv — COMMENTED** [review]
`disableUpsert` is a bit confusing. Initially it led me to believe that it is actually going to disable upsert operation, which led me to question why the read path should have the option to disable upsert in the write path.

Upon reading the changes, I feel a better name might be something along the lines of `skipUpsert`, `ignoreUpsert` or `skipUpsertedDocIds`, for better readability.

**Comment by yupeng9** [comment]
> `disableUpsert` is a bit confusing. Initially it led me to believe that it is actually going to disable upsert operation, which led me to question why the read path should have the option to disable upsert in the write path.
> 
> Upon reading the changes, I feel a better name might be something along the lines of `skipUpsert`, `ignoreUpsert` or `skipUpsertedDocIds`, for better readability.

Makes sense. I prefer `skipUpsert`.

File: pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptions.java [hunk 1]
```diff
@@ -64,6 +67,10 @@ public boolean isPreserveType() {
     return _preserveType;
   }
 
+  public boolean isUpsertSkipped() {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptions.java:70]: (nit) `isSkipUpsert()` to be consistent with the variable name?

**Review by Jackie-Jiang — APPROVED** [review]
LGTM with minor comments

**Pull request merged**